### DISCUSSION TITLE
Credentials "authorize" method is not triggered NextJS 13.4.13-canary.0

### DISCUSSION THREAD

**Discussion by tpetersons** [nextauthjs/next-auth#8369] [Help]
### Question 💬

Hello,

Does anyone else have issue with NextJS 13.4.13 version while using Credentials provider?

Everything works fine with NextJS 13.4.12, but as soon as I update the version, I cannot successfully authorise my user anymore.

I've added console logs to "authorize" method in `/app/api/auth/[...nextauth]/route.ts` and they are not being shown in console, meaning the method never gets triggered.

Whereas if I downgrade to 13.4.12. I receive logs immediately.

I am using prisma adapter but I dont think that matters since I am not getting that far in code.

### How to reproduce ☕️

This is my custom login component:

```
signIn('credentials', { email, password, redirect: false })
			.then(async (response: any) => {
				if (response.status === 200 && !response?.error) {
					window.location.replace('/builder');
				} else {
					setError('Provided credentials are wrong.');
					setLoading(false);
				}
			})
			.catch(() => {
				setError('Something went wrong.');
				setLoading(false);
			});
```

And this is how route.ts file looks like:

```
import { verify } from 'argon2';
import { randomBytes, randomUUID } from 'crypto';
import NextAuth, { AuthOptions } from 'next-auth';
import { PrismaAdapter } from '@next-auth/prisma-adapter';
import prisma from '../../../../../prisma/prisma';
import CredentialsProvider from 'next-auth/providers/credentials';

const authOptions = {
	providers: [
		CredentialsProvider({
			name: 'Credentials',
			credentials: {
				username: { label: 'Email', type: 'email' },
				password: { label: 'Password', type: 'password' },
			},
			async authorize(credentials: any) {
				const user = await prisma.user.findFirst({
					where: { email: credentials.email.toLowerCase() },
				});

				if (!user) {
					throw new Error('Provided credentials are invalid.');
				}

				const isValidPassword = await verify(
					user.password as string,
					credentials.password
				);

				if (!isValidPassword) {
					throw new Error('Provided credentials are invalid.');
				}

				return {
					id: user.id,
					username: user.username,
					email: user.email,
					image: user.image,
				};
			},
		}),
	],
	adapter: PrismaAdapter(prisma),
	secret: process.env.SECRET,
	session: {
		// Choose how you want to save the user session.
		// The default is `"jwt"`, an encrypted JWT (JWE) stored in the session cookie.
		// If you use an `adapter` however, we default it to `"database"` instead.
		// You can still force a JWT session by explicitly defining `"jwt"`.
		// When using `"database"`, the session cookie will only contain a `sessionToken` value,
		// which is used to look up the session in the database.
		strategy: 'jwt',

		// Seconds - How long until an idle session expires and is no longer valid.
		maxAge: 15 * 24 * 30 * 60, // 30 days

		// Seconds - Throttle how frequently to write to database to extend a session.
		// Use it to limit write operations. Set to 0 to always update the database.
		// Note: This option is ignored if using JSON Web Tokens
		updateAge: 24 * 60 * 60, // 24 hours

		// The session token is usually either a random UUID or string, however if you
		// need a more customized session token string, you can define your own generate function.
		generateSessionToken: () => {
			return randomUUID?.() ?? randomBytes(32).toString('hex');
		},
	},
	pages: {
		signIn: '/login',
	},
	events: {
		async signIn(message) {
			/* on successful sign in */
		},
		async signOut(message) {
			/* on signout */
		},
		async createUser(message) {
			/* user created */
		},
		async updateUser(message) {
			/* user updated - e.g. their email was verified */
		},
		async linkAccount(message) {
			/* account (e.g. Twitter) linked to a user */
		},
		async session(message) {
			/* session is active */
		},
	},
	callbacks: {
		async signIn({ user, account, profile, email, credentials }) {
			return true;
		},
		async redirect({ url, baseUrl }) {
			return baseUrl;
		},
		async session({ session, token, user }: any) {
			if (token) {
				session.id = token.id;
			}

			return session;
		},
		async jwt({ token, user, account, profile, isNewUser }) {
			if (user) {
				token.id = user.id;
				token.email = user.email;
			}

			return token;
		},
	},
	jwt: {
		maxAge: 15 * 24 * 30 * 60, // 15 days
	},
	debug: process.env.NODE_ENV !== 'production',
} as AuthOptions;

const handler = NextAuth(authOptions);
export { handler as GET, handler as POST, authOptions };

```

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by benjaminwaterlot** [comment]
@tpetersons I've met problems that might have the same cause as you when updating Next.js from `13.4.12` to `13.4.13-canary.0` and any newer version.


In my case, it looks like every calls to the `GET` / `POST` `next-auth` route handler is re-written to `https://localhost:7777` instead of `http://localhost:7777`, which means every call was failing with errors like `ERR_SSL_PROTOCOL_ERROR` (my `NEXTAUTH_URL` env variable is fine - it seems the endpoint systematically rewrites the request to another URL).

I witnessed the problem for the Google OAuth provider, and haven't tried the credentials provider.


**→ Can you check in your browser network tab, looking for failed calls to your auth endpoints ?**


In my case, I searched for hours and was not able to find a solution. 

The problem only exists in development server (deployed envs worked as expected), and goes away when I use `local-ssl-proxy` to proxy my dev server to using SSL and a `https` url (although that approach seems to bring other problems).

I'm trying to make a reproduction for @balazsorban44 , but have been unable to do so and didn't open an issue for that reason.

**Reply by balazsorban44** [reply]
I can have a closer look when you managed to create one! Please open it as a bug report. Thanks!

**Reply by benjaminwaterlot** [reply]
Thanks, I'll try to do a reproduction then 🙏

**Reply by benjaminwaterlot** [reply]
Finally got it! 😍

So it's probably not related to the issue of this discussion, but having the `VERCEL` env variable set to any truthy value causes the issue in dev, starting `13.4.13-canary.0`. 

This env variable is set automatically by the Vercel CLI even in development, so it should not be used to discriminate if we're running on a Vercel deployed env.

Issue: https://github.com/nextauthjs/next-auth/issues/8385

**Reply by hirehamir** [reply]
I just ran into the `ERR_SSL_PROTOCOL_ERROR` issue a few minutes ago and fixed it by commenting out `VERCEL="1"` in `.env`. @benjaminwaterlot, your comment helped me resolve this issue. Thank you!
